WordPress/src/test/java/org/wordpress/android/ui/mysite/QuickStartRepositoryTest.kt

```diff
@@ -105,26 +110,59 @@ class QuickStartRepositoryTest : BaseUnitTest() {
     @Test
-    fun `refresh shows completion message and removes card if all tasks of a same type have been completed`() = test {
-        initStore()
+    fun `given dynamic card enabled + same type tasks done, when refresh started, then completion msg shown`() =
+            test {
+                whenever(quickStartDynamicCardsFeatureConfig.isEnabled()).thenReturn(true)
+                initStore()
 
-        val completionMessage = "All tasks completed!"
+                triggerQSRefreshAfterSameTypeTasksAreComplete()
 
-        whenever(selectedSiteRepository.getSelectedSite()).thenReturn(site)
-        whenever(quickStartUtils.isEveryQuickStartTaskDoneForType(siteId, GROW)).thenReturn(true)
-        whenever(resourceProvider.getString(any())).thenReturn(completionMessage)
-        whenever(htmlCompat.fromHtml(completionMessage)).thenReturn(completionMessage)
+                assertThat(snackbars).containsOnly(SnackbarMessageHolder(UiStringText(ALL_TASKS_COMPLETED_MESSAGE)))
+            }
 
-        val task = PUBLISH_POST
-        quickStartRepository.setActiveTask(task)
-        quickStartRepository.completeTask(task)
-        quickStartRepository.refresh()
+    @Test
+    fun `given dynamic card disabled + same type tasks done, when refresh started, then completion msg not shown`() =
+            test {
+                initStore()
+
+                triggerQSRefreshAfterSameTypeTasksAreComplete()
+
+                assertThat(snackbars).doesNotContain(SnackbarMessageHolder(UiStringText(ALL_TASKS_COMPLETED_MESSAGE)))
+            }
+
+    @Test
+    fun `given dynamic card enabled + same type tasks done, when refresh started, then dynamic card removed`() = test {
+        whenever(quickStartDynamicCardsFeatureConfig.isEnabled()).thenReturn(true)
+        initStore()
 
-        assertThat(snackbars).containsOnly(SnackbarMessageHolder(UiStringText(completionMessage)))
+        triggerQSRefreshAfterSameTypeTasksAreComplete()
 
         verify(dynamicCardStore).removeCard(siteId, GROW_QUICK_START)
     }
 
+    @Test
+    fun `given dynamic card disabled + same type tasks done, when refresh started, then dynamic card not removed`() =
+            test {
+                initStore()
+
+                triggerQSRefreshAfterSameTypeTasksAreComplete()
+
+                verify(dynamicCardStore, never()).removeCard(siteId, GROW_QUICK_START)
+            }
+
+    @Test
+    fun `given dynamic card disabled + same type tasks done, when refresh started, then both task types exists`() =
+            test {
+                whenever(quickStartDynamicCardsFeatureConfig.isEnabled()).thenReturn(false)
+                initStore()
+
+                triggerQSRefreshAfterSameTypeTasksAreComplete()
+                var result: QuickStartUpdate? = null
+                quickStartRepository.buildSource(testScope(), siteId).observeForever { result = it }
+
+                assertThat(result?.categories?.map { it.taskType }).isEqualTo(listOf(CUSTOMIZE, GROW))
```

Comment:
Minor (🔍): Will the below be equivalent to this assertion here.

So, instead of setting the `result` locally:
```
var result: QuickStartUpdate? = null
quickStartRepository.buildSource(testScope(), siteId).observeForever { result = it }

assertThat(result?.categories?.map { it.taskType }).isEqualTo(listOf(CUSTOMIZE, GROW))
```

Can we do just this and reuse the field `result`:

```
assertThat(result.last().categories.map { it.taskType }).isEqualTo(listOf(CUSTOMIZE, GROW))
```

Reply:
Oh yes, it is present in the test setup and can be reused, thanks for noticing it 🙇‍♀️.
In bce72c1.